R/sql_methods.R

```diff
@@ -50,13 +50,22 @@ read_sql <- function(sql, dbname = NULL, schemaname = NULL){
  # Unlisting list column types. Also substituting NULL values to NA to remain consistent.
  for (i in seq(1,ncol(result))){
    if (typeof(result[,i]) == "list"){
-     if (class(result[,i][[1]])[1] == "datetime.date"){
-       
-       # returning dates in posixct format if stored as dates.
+     
+     # finding the first not null element in the list to determine its class. If it is python's datetime.date, returning it as Date.
+     k = 1
+     if (is.null(result[,i][[1]])) {
+       while (is.null(result[,i][[k]])) {
+         k = k +1
+       }
+     }
```

Comment:
I think this could be shortened by dropping the first if.

Also please add a check to make sure that k <= length of result[,i], otherwise you will get an error for all NULL columns